mllib/src/main/scala/org/apache/spark/ml/optim/WeightedLeastSquares.scala

```diff
@@ -94,8 +110,7 @@ private[ml] class WeightedLeastSquares(
       if (standardizeFeatures) {
         lambda *= aVar(j - 2)
       }
-      if (standardizeLabel) {
-        // TODO: handle the case when bStd = 0
+      if (standardizeLabel && bStd != 0) {
```

Comment:
It's interesting to see that when regularization is zero, with/without standardization on labels or features will not change the solution of Linear Regression which you can experiment. 

As a result, the only issue that the model will be non-interpretable will be `yStd` is zeo and `regParam` is non-zero. You can have a `require` there with proper message. I think logging a warning will be probably very easy for users to ignore. Thanks.